tests/provision/provisioning_rules.go

```diff
@@ -0,0 +1,1708 @@
+apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: vps-seq-rule
+spec:
+  replicaAntiAffinity:
+  - enforcement: required
+    topologyKey: failure-domain.beta.kubernetes.io/px_zone
+  replicaAffinity:
+  - enforcement: required
+    matchExpressions:
+      - key: ioprofile
+        operator: In
+        values:
+          - "REGULAR"
+---
+apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: vps-aggr-rule
+spec:
+  volumeAntiAffinity:
+  - enforcement: required
+    matchExpressions:
+      - key: voltype
+        operator: In
+        values:
+          - "mysql-data"
+  replicaAffinity:
+  - enforcement: required
+    matchExpressions:
+      - key: media_type
+        operator: In
+        values:
+          - "SSD"
+  volumeAffinity:
+  - enforcement: required
+    matchExpressions:
+      - key: voltype
+        operator: In
+        values:
+          - "mysql-seq"`
+	return vpsSpec
+}
+
+func (v *vpscase23) CleanVps() {
+	logrus.Infof("Cleanup test case context for: %v", v.name)
+}
+
+//#---- Case 24 ----T864240  Verify Replica Anti-Affinity with topology keys
+// With few nodes labeled
+type vpscase24 struct {
+	//Case description
+	name string
+	// Enabled
+	enabled bool
+}
+
+func (v *vpscase24) GetLabels() ([]labelDict, int) {
+
+	lbldata := []labelDict{}
+	node1lbl := labelDict{"failure-domain.beta.kubernetes.io/zone": "east"}  //, "failure-domain.beta.kubernetes.io/region": "usa"}
+	node2lbl := labelDict{"failure-domain.beta.kubernetes.io/zone": "east"}  //, "failure-domain.beta.kubernetes.io/region": "usa"}
+	node3lbl := labelDict{"failure-domain.beta.kubernetes.io/zone": "west"}  //, "failure-domain.beta.kubernetes.io/region": "asia"}
+	node4lbl := labelDict{"failure-domain.beta.kubernetes.io/zone": "west"}  //, "failure-domain.beta.kubernetes.io/region": "asia"}
+	node5lbl := labelDict{"failure-domain.beta.kubernetes.io/zone": "south"} //, "failure-domain.beta.kubernetes.io/region": "eu"}
+	node6lbl := labelDict{"failure-domain.beta.kubernetes.io/zone": "south"} //, "failure-domain.beta.kubernetes.io/region": "eu"}
+	lbldata = append(lbldata, node1lbl, node2lbl, node3lbl, node4lbl, node5lbl, node6lbl)
+	return lbldata, 1
+}
+
+//StorageClass placement_strategy mapping
+func (v *vpscase24) GetScStrategyMap() map[string]string {
+	return map[string]string{"placement-1": "placement-1", "placement-2": "placement-1", "placement-3": ""}
+}
+
+func (v *vpscase24) GetSpec() string {
+
+	var vpsSpec string
+	vpsSpec = `apiVersion: portworx.io/v1beta2
+kind: VolumePlacementStrategy
+metadata:
+  name: placement-1
+spec:
+  replicaAntiAffinity:
+  - enforcement: required
+    topologyKey: failure-domain.beta.kubernetes.io/zone`
+	return vpsSpec
+}
+
+func (v *vpscase24) CleanVps() {
+	logrus.Infof("Cleanup test case context for: %v", v.name)
+}
+
+// Test case inits
+//
+
+/*
+ *
+ *     Replica  Affinity and Anti-Affinity related test cases init
+ *
+ */
+///*
+func init() {
```

Comment:
is there any specific reason why we have init functions define so many times?

Reply:
The init function is adding  each test case to the list of test cases to be tested. 
This will changed, if the changes as per your comment below is incorporated
`instead of adding this put the rules in an indexed list like var rules []map[string]interface{}, so you can use return rules[cat]`

Reply:
Modified the init function, please review it again